### PR TITLE
Remove Python 2.2 support code

### PR DESCRIPTION
**Pull request by chasinglogic**
This is a code cleanup and it should be slightly faster.

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

**Comment by bdbaddog** [comment]
This code drops the current functionality of not changing the order of the items in the list...
set doesn't guarantee no change in order?

**Comment by bdbaddog** [comment]
If you wait til 4.0.0 (and py3 only support) you can use dict() which maintains insertion order of keys for traversal. Though likely larger memory footprint.

**Comment by chasinglogic** [comment]
Does order matter for this data structure? Nothing in the code that uses this that I found, or in the original code itself stated that the insertion order matters for this.

AFAICT it's only used for lists of targets and sources for execution (which at that point it's treated just like a flat list, no priority) and for the C Scanner which also doesn't matter it seems to me.

**Comment by chasinglogic** [comment]
Probably worth adding that this still correctly builds and rebuilds in MongoDB's CI (and on my laptop)

**Comment by mwichmann** [comment]
if order-preserving actually matters, then this is fast for `uniquer_hashables`: `return list(dict.fromkeys(seq))` (slight disadvantage: needs new dicts, so not-3.5).  But as far as I can see this function is only used by `FindInstalledFiles` which is only used by packaging tools, where it presumably isn't needed. The submitted line is, as far as I know, much faster. Indeed, that function could just be dropped from Util and the code used inline in the one place it's used.

**Comment by chasinglogic** [comment]
After looking at the call sites I realized that SConsign tracks order of dependencies for tracking out of date-ness. I think this means the bottom line should be anywhere that's using a UniqueList that cares about order should probably call sorted? Otherwise the order isn't guaranteed. Insertion does not have to happen in the same order in SCons (we've seen this cause rebuilds actually at Mongo)

**Comment by bdbaddog** [comment]
In this logic it's expecting the insertion order to be maintained so changing that would likely have unintented consequences and should be avoided.   

The issue of sorting (for example sources specified to a library) is separate.

File: src/engine/SCons/Util.py [hunk 1]
```diff
@@ -1175,44 +1175,12 @@ def unique(s):
...
 def uniquer_hashables(seq):
-    seen = {}
-    result = []
-    for item in seq:
-        #if not item in seen:
-        if item not in seen:
-            seen[item] = 1
-            result.append(item)
-    return result
+    return list(set(seq))
```
Comment (bdbaddog) [inline src/engine/SCons/Util.py:1183]: How about this: http://code.activestate.com/recipes/52560/#c3

Comment (mwichmann) [reply src/engine/SCons/Util.py:1183]: Things have evolved, there aren't a lot of 17-year old Python recipes that are still the best answer...

**Pull request closed without merging**